File: benchmarks/subjson/java/StringCopies.java
package subjson.perftests;

public class StringCopies
{
    public static String copyStringSBChars(String s)
    {
        final int len = s.length();
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < len; i++) {
            sb.append(s.charAt(i));
        }

        return sb.toString();
    }

    public static String copyStringSBStr(String s)
    {
        StringBuilder sb = new StringBuilder();
        sb.append(s);

        return sb.toString();
    }

    public static String copyStringCharArray(char[] s)
    {
        return new String(s);
    }
}

File: benchmarks/subjson/benchmarks/string_copies.clj
(ns subjson.benchmarks.string-copies
  (:use perforate.core)
  (:require [clojure.java.io :as io])
  (:import [subjson.perftests StringCopies]))

(defgoal string-copy "Create a copy of a string")

(defcase string-copy :stringbuilder-chars
  []
  (let [s "abcdefghijklmnopqrstuvwxyz"]
    (dotimes [_ 1000] (StringCopies/copyStringSBChars s))))

(defcase string-copy :stringbuilder-str
  []
  (let [s "abcdefghijklmnopqrstuvwxyz"]
    (dotimes [_ 1000] (StringCopies/copyStringSBStr s))))


(defcase string-copy :arraycopy
  []
  (let [s (.toCharArray "abcdefghijklmnopqrstuvwxyz")]
    (dotimes [_ 1000] (StringCopies/copyStringCharArray s))))

